src/engine/GraphStoreProtocol.cpp

```diff
@@ -68,13 +69,31 @@ std::vector<SparqlTripleSimpleWithGraph> GraphStoreProtocol::convertTriples(
 ParsedQuery GraphStoreProtocol::transformGet(const GraphOrDefault& graph) {
   // Construct the parsed query from its short equivalent SPARQL Update string.
   // This is easier and also provides e.g. the `_originalString` field.
-  std::string query;
-  if (const auto* iri =
-          std::get_if<ad_utility::triple_component::Iri>(&graph)) {
-    query = absl::StrCat("CONSTRUCT { ?s ?p ?o } WHERE { GRAPH ",
-                         iri->toStringRepresentation(), " { ?s ?p ?o } }");
-  } else {
-    query = "CONSTRUCT { ?s ?p ?o } WHERE { ?s ?p ?o }";
-  }
+  auto query = [&graph]() -> std::string {
+    if (const auto* iri =
+            std::get_if<ad_utility::triple_component::Iri>(&graph)) {
+      return absl::StrCat("CONSTRUCT { ?s ?p ?o } WHERE { GRAPH ",
+                          iri->toStringRepresentation(), " { ?s ?p ?o } }");
+    } else {
+      return "CONSTRUCT { ?s ?p ?o } WHERE { ?s ?p ?o }";
+    }
+  }();
   return SparqlParser::parseQuery(query);
 }
+
+// ____________________________________________________________________________
+ParsedQuery GraphStoreProtocol::transformDelete(const GraphOrDefault& graph) {
+  // Construct the parsed update from its short equivalent SPARQL Update string.
+  // This is easier and also provides e.g. the `_originalString` field.
+  auto update = [&graph]() -> std::string {
+    if (const auto* iri =
+            std::get_if<ad_utility::triple_component::Iri>(&graph)) {
+      return absl::StrCat("DROP GRAPH ", iri->toStringRepresentation());
+    } else {
+      return "DROP DEFAULT";
+    }
+  }();
+  auto updates = SparqlParser::parseUpdate(update);
```

Comment:
Same comment as above (unneded copy).

Reply:
And maybe you can do `return ad_utility::getSingleElement(parseUpdate(getUpdate()))`
(in `util/Iterator.h`.